### PR TITLE
✨[Amp story] [Page attachments] Adding support for title & cta-text attributes 

### LINKED ISSUE #33724: [Page attachments] [Outlink] Support cta-text and title attributes
Allow cta-text and title and update the documentation, while still supporting data-cta-text and data-title to be backward compatible

### PR DESCRIPTION
**Pull request by raxsha**
Previously, these two attributes were `data-title` and `data-cta-text` but we're moving forward without `data-` prepended attribute names for consistency.

`data-title` & `data-cta-text` will not be deprecated; they will be silently supported, while we update documentation & code paths to prioritize `title` & `cta-text` attributes

Fixes #33724 and #34050.

### CODE REVIEW

**Comment by newmuis** [comment]
FYI the `title` attribute has [another meaning](https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/title) in HTML, so you may need to do some extra work to keep it from showing a tooltip on hover on desktop.  We also [do this for amp-story](https://github.com/ampproject/amphtml/blob/164bec0b8b9a38345b1cb216b130277eff3b214e/extensions/amp-story/1.0/amp-story.js#L444) (aka we remove the attribute after we've copied the data elsewhere).

**Comment by raxsha** [comment]
> FYI the `title` attribute has [another meaning](https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/title) in HTML, so you may need to do some extra work to keep it from showing a tooltip on hover on desktop. We also [do this for amp-story](https://github.com/ampproject/amphtml/blob/164bec0b8b9a38345b1cb216b130277eff3b214e/extensions/amp-story/1.0/amp-story.js#L444) (aka we remove the attribute after we've copied the data elsewhere).

I've copied the value of the `title` attribute to a `data-title` attribute (this will override `data-title` when both exist, but that's intended since we'll look for `title` first always) and removed the `title` attribute on `amp-story-page-attachment`. You can see a comment that says `// To prevent 'title' attribute from being used by browser, copy value to 'data-title' and remove.` in the code.

**Review by MichaelRybak — APPROVED** [review]

File: examples/amp-story/attachment.html [hunk 1]
```diff
@@ -170,13 +170,13 @@
...
         <amp-story-page-attachment
-          data-title="Link Description"
+          title="Link Description"
           href="https://www.google.com"
           layout="nodisplay">
         </amp-story-page-attachment>
       </amp-story-page>
 
-      <amp-story-page id="outlink-custo-icon-custom-text">
+      <amp-story-page id="outlink-custom-icon-custom-text">
```
Comment (processprocess) [inline examples/amp-story/attachment.html:179]: Thanks for fixing my typo :)

**Review by processprocess — APPROVED** [review]
Looks good to me. Thanks for updating the docs in this PR too :)

File: extensions/amp-story/validator-amp-story.protoascii [hunk 2]
```diff
@@ -859,6 +859,12 @@ tags: {  # <amp-story-page-attachment> with href
     value: "dark"
     value: "light"
   }
+  attrs: {
+    name: "cta-text"
+  }
+  attrs: {
```
Comment (mszylkowski) [inline extensions/amp-story/validator-amp-story.protoascii:865]: Is it worth to make `title` and `data-title` exclusive (and same thing with `cta-text` and `data-cta-text`)?

**Review by gmajoulet — APPROVED** [review]

File: extensions/amp-story/validator-amp-story.protoascii (hunk 2 above)
Comment (raxsha) [reply extensions/amp-story/validator-amp-story.protoascii:865]: Doesn't look like that's an option, without making one of them mandatory. The code overrides data- attributes when title or cta-text are set as well

**Review by mszylkowski — APPROVED** [review]

**Pull request merged**